rat-trap.patch

```diff
@@ -0,0 +1,321 @@
++	private ["_allPlayerPositions", "_allPlayerPositionsTemp", "_activeVehiclesAndGroup", "_vehicleGroup", "_spawnSegment", "_vehicle", "_group", "_result", "_possibleVehicles", "_vehicleType", "_vehiclesCrew", "_skill", "_minDistance", "_tries", "_trafficLocation"];
+ 	private ["_currentEntityNo", "_vehicleVarName", "_tempVehiclesAndGroup", "_deletedVehiclesCount", "_firstIteration", "_roadSegments", "_destinationSegment", "_destinationPos", "_direction"];
+ 	private ["_roadSegmentDirection", "_testDirection", "_facingAway", "_posX", "_posY", "_pos", "_currentInstanceIndex"];
+ 	private ["_fnc_OnSpawnVehicle", "_fnc_OnRemoveVehicle", "_fnc_FindSpawnSegment"];
+@@ -163,11 +200,13 @@ ENGIMA_TRAFFIC_StartTraffic = {
+ 	_fnc_OnRemoveVehicle = [_this, "ON_REMOVE_CALLBACK", {}] call ENGIMA_TRAFFIC_GetParamValue;
+ 	_debug = [_this, "DEBUG", false] call ENGIMA_TRAFFIC_GetParamValue;
+ 
++
++
+ 	if (_areaMarkerName != "" && _hideAreaMarker) then {
+ 		_areaMarkerName setMarkerAlpha 0;
+ 	};
+ 
+-	sleep random 1;
++	sleep (random 1);
+ 	ENGIMA_TRAFFIC_instanceIndex = ENGIMA_TRAFFIC_instanceIndex + 1;
+ 	_currentInstanceIndex = ENGIMA_TRAFFIC_instanceIndex;
+ 
+@@ -179,7 +218,7 @@ ENGIMA_TRAFFIC_StartTraffic = {
+ 
+ 	_fnc_FindSpawnSegment = {
+ 	    private ["_currentInstanceIndex", "_allPlayerPositions", "_minSpawnDistance", "_maxSpawnDistance", "_activeVehiclesAndGroup"];
+-	    private ["_insideMarker", "_areaMarkerName", "_refPlayerPos", "_roadSegments", "_roadSegment", "_isOk", "_tries", "_result", "_spawnDistanceDiff", "_refPosX", "_refPosY", "_dir", "_tooFarAwayFromAll", "_tooClose", "_tooCloseToAnotherVehicle"];
++	    private ["_isRoad", "_insideMarker", "_areaMarkerName", "_refPlayerPos", "_roadSegments", "_roadSegment", "_isOk", "_tries", "_result", "_spawnDistanceDiff", "_refPosX", "_refPosY", "_dir", "_tooFarAwayFromAll", "_tooClose", "_tooCloseToAnotherVehicle"];
+ 
+ 		_currentInstanceIndex = _this select 0;
+ 	    _allPlayerPositions = _this select 1;
+@@ -212,6 +251,11 @@ ENGIMA_TRAFFIC_StartTraffic = {
+ 	            _tooClose = false;
+                 _insideMarker = true;
+                 _tooCloseToAnotherVehicle = false;
++                _isRoad = false;
++
++                if (isOnRoad (ASLToAGL getPosASL _roadSegment)) then {
++                	_isRoad = true;
++                };
+ 
+                 if (_areaMarkerName != "" && !([getPos _roadSegment, _areaMarkerName] call ENGIMA_TRAFFIC_PositionIsInsideMarker)) then {
+                 	_insideMarker = false;
+@@ -250,7 +294,7 @@ ENGIMA_TRAFFIC_StartTraffic = {
+ 
+ 	            _isOk = true;
+ 
+-	            if (_tooClose || _tooFarAwayFromAll || _tooCloseToAnotherVehicle || !_insideMarker) then {
++	            if (_tooClose || _tooFarAwayFromAll || _tooCloseToAnotherVehicle || !_insideMarker || !_isRoad) then {
```

Comment:
the whole concept of checking for roads didnt work out in the end, as waypoints/driving routes are set by the engine. now its better to use the new driving ai commands anyway